flytekit/core/type_engine.py

```diff
@@ -437,7 +447,29 @@ def to_literal_type(cls, python_type: Type) -> LiteralType:
         Converts a python type into a flyte specific ``LiteralType``
         """
         transformer = cls.get_transformer(python_type)
-        return transformer.get_literal_type(python_type)
+        res = transformer.get_literal_type(python_type)
+        data = None
+        if typing.get_origin(python_type) is typing.Annotated:
+            for x in typing.get_args(python_type)[1:]:
+                if not isinstance(x, FlyteAnnotation):
+                    continue
+                if x.data.get("__consumed", False):
+                    continue
+                data = x.data
+                x.data["__consumed"] = True
+        if data is not None:
+            idl_type_annotation = _annotation_model(annotations=data)
+            return LiteralType(
```

Comment:
these are python objects - it should be safe to just attach the annotation result to the existing `res` object right?

Reply:
well no, they need to be serialized into proto correctly given the new message created to suppport annotations
https://github.com/flyteorg/flyteidl/pull/232/files#diff-da0cd41ae99f181280058bc79cbc3f3565a22039fbefcbe5f191bc2780d72735R68